sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsServiceSerializer.cs

```diff
@@ -558,6 +558,8 @@ private static RecognizeLinkedEntitiesResultCollection ReadLinkedEntityResultCol
 
             foreach (var error in ReadDocumentErrors(root))
             {
+                if (string.IsNullOrEmpty(error.Id))
```

Comment:
Is the serializer the right place to throw the exception?  It feels like this could imply it's a serialization exception if someone looked at the exception.  I was expecting to see it in TextAnalyticsClient...  As long as it's clear what the exception is and it's thrown in the right client method, it may be ok.  It'll make it harder to understand what exceptions are thrown by what methods just by looking through the client code ... not sure if that's something we want.

Reply:
I know... it looks bad.. my though here was, ok this is temporary o might as well, but you are right.
I will go back and look for another approach

Reply:
That's a good point about temporary - it occurred to me later as well.  I wouldn't spend too long coming up with another approach, given this.